lib/commands/execute.js

```diff
@@ -0,0 +1,102 @@
+import { exec } from 'teen_process';
+import _ from 'lodash';
+import log from '../logger';
+import { fs, tempDir } from 'appium-support';
+import path from 'path';
+
+const SYSTEM_SHELL_FEATURE = 'system_shell';
+
+const commands = {};
+
+/**
+ * @typedef {Object} ExecOptions
+ * @property {?string} interpreter - Full path to the command line interpreter binary.
+ * The current interpreter (`$SHELL`) or `/bin/bash` is used by default
+ * @property {?boolean} throwOnFail [false] - Whether to throw an exception if
+ * the given script has returned non-zero exit code
+ * @property {?number} timeout [20000] - The default timeout for the script execution.
+ * @property {?Object} env [process.env] - Additional environment variables for
+ * the shell script
+ */
+
+/**
+ * @typedef {Object} ExecResult
+ * @property {string} stdout - Script stdout
+ * @property {string} stderr - Script stderr
+ * @property {number} code - Script return code. It will never be other
+ * than zero if `throwOnFail` option is set to `true`
+ */
+
+/**
+ * Executes the given shell script if the `system_shell`
+ * server feature is enabled. The command blocks until
+ * the script finishes its execution or its timeout expires.
+ *
+ * @param {!string} script - The actual shell script to execute.
+ * This should be a valid script snippet.
+ * @param {?ExecOptions} args
+ * @return {ExecResult} - The result of the script execution
+ * @throws {Error} If there was a problem during command execution
+ */
+commands.execute = async function execute (script, args) {
+  this.ensureFeatureEnabled(SYSTEM_SHELL_FEATURE);
```

Comment:
Will we add this feature flag in readme like other modules?

Reply:
Sorry, I'm sometimes being lazy with documenting all the stuff. I've tried to provide accurate docstrings in the source though ;)

Reply:
@mykola-mokhnach can you please provide an example of how to actually execute a simple shell command once this feature has been enabled when starting appium server? im unsure how to use with my driver? I'm using C# fyi. Appreciate any help. Thanks :)